src/structure.js

```diff
@@ -68,6 +69,34 @@ Structure.prototype.cursor = function (path) {
   return Cursor.from(self.current, path, changeListener);
 };
 
+Structure.prototype.reference = function (path, fn) {
+  var self = this;
+  fn = fn || function () { };
+
+  var cursor = this.cursor(path);
+  var changeListener = function (newRoot, oldRoot, changedPath) {
+    fn(newRoot, oldRoot, path);
+    cursor = self.cursor(path);
+  };
+  this.on('swap', changeListener);
+
+  return {
+    onChange: function (newFn) {
```

Comment:
To clarify, `fn` reference still exists when calling `onChange` after calling `unsubscribe`.